File: python/problem27.py
from primes import isprime

amax = 1000
bmax = 1000

def consecutive_primes(a,b):
    # n^2 + (a*n) + b

    n = 0
    while True:
        p = (n * n) + (a * n) + b
        if not isprime(p):
            return n

        n += 1

results = [(a, b, consecutive_primes(a,b))
            for a in range(amax * -1, amax)
            for b in range(bmax * -1, bmax)]

n, a, b = sorted(((n,a,b) for a,b,n in results if n > 1),
                reverse=True)[0]

print "a = {0}, b = {1}, n = {2}".format(a,b,n)

File: cpp/problem27.cpp
#include <stdio.h>
#include <stdlib.h>

#include "primes.h"

using namespace std;

#define uint unsigned int

int amax = 1000;
int bmax = 1000;

int consecutive_primes(int a, int b)
{
    int n = 0;
    int result = 0;

    while(true)
    {
        result = (n * n) + (a * n) + b;

        if (!isprime(result))
        {
            return n;
        }

        n++;
    }
}

int main(int argc, char* argv[])
{
    int largest_n = 0;
    int largest_a = 0;
    int largest_b = 0;
    int n = 0;

    for (int a = amax * -1; a <= amax; a++)
    {
        for (int b = bmax * -1; b <= bmax; b++)
        {
            n = consecutive_primes(a, b);

            if (n > largest_n)
            {
                largest_n = n;
                largest_a = a;
                largest_b = b;
            }
        }
    }

	printf("a = %d, b = %d, n = %d\n", largest_a, largest_b, largest_n);
	return 0;
}

